test/mobile/test_lite_script_module.py

```diff
@@ -431,5 +445,109 @@ def forward(self, val: int, x, y, w):
         self.assertTrue('test_lite_script_module.py\", line {}'.format(lineno + 8) in error_message)
         self.assertTrue('top(FooTest5)' in error_message)
 
+
+class TestLiteScriptQuantizedModule(QuantizationTestCase):
+
+    def _compare_script_and_mobile(self,
+                                   model_class: Type[torch.nn.Module],
+                                   input: torch.Tensor,
+                                   **kwargs):
+        qengine = "qnnpack"
+        with override_quantized_engine(qengine):
+            qconfig = torch.quantization.get_default_qconfig(qengine)
+            model = model_class(**kwargs)
+            # model.qconfig = qconfig
+            # model = prepare(model)
+
+            # test_only_eval_fn(model, self.calib_data)
+            # model = convert(model)
+            model = quantize(model, test_only_eval_fn, [self.calib_data])
+
+            script_module = torch.jit.script(model)
+            script_module_result = script_module(input)
+
+            max_retry = 5
+            for retry in range(1, max_retry + 1):
+                # retires `max_retry` times; breaks iff succeeds else throws exception
+                try:
+                    buffer = io.BytesIO(script_module._save_to_buffer_for_lite_interpreter())
+                    buffer.seek(0)
+                    mobile_module = _load_for_lite_interpreter(buffer)
+
+                    mobile_module_result = mobile_module(input)
+
+                    torch.testing.assert_allclose(script_module_result, mobile_module_result)
+                    mobile_module_forward_result = mobile_module.forward(input)
+                    torch.testing.assert_allclose(script_module_result, mobile_module_forward_result)
+
+                    mobile_module_run_method_result = mobile_module.run_method("forward", input)
+                    torch.testing.assert_allclose(script_module_result, mobile_module_run_method_result)
+                except AssertionError as e:
+                    if retry == max_retry:
+                        raise e
+                    else:
+                        continue
+                break
+
+
+
+    def test_single_layer(self):
+        input = torch.rand(2, 5, dtype=torch.float)
+        self._compare_script_and_mobile(model_class=AnnotatedSingleLayerLinearModel, input=input, qengine="qnnpack")
+
+    def test_two_layer(self):
+        input = torch.rand(2, 5, dtype=torch.float)
+        self._compare_script_and_mobile(model_class=TwoLayerLinearModel, input=input)
+
+    def test_annotated_nested(self):
+        input = torch.rand(2, 5, dtype=torch.float)
+        self._compare_script_and_mobile(model_class=AnnotatedNestedModel, input=input, qengine="qnnpack")
+
+    def test_quantization_example(self):
```

Comment:
Why aren't we using `_compare_script_and_mobile` for this test as well?

Reply:
you are right, I will try and split `_compare_script_and_mobile` into two functions. One to create a module object and the other to compare output tensors (script and lite).